src/opera/pge/base_pge.py

```diff
@@ -203,14 +206,40 @@ class and its inheritors, and as such, this class assumes access to the
     def __init__(self):
         self.name = None
         self.logger = None
+        self.runconfig_path = None
+        self.runconfig = RunConfig(self.runconfig_path)
+        self.catalog_metadata = {}
 
     def _run_sas_qa_executable(self):
         # TODO
         pass
 
     def _create_catalog_metadata(self):
-        # TODO
-        pass
+        """Returns the catalog metadata as a Python dictionary"""
+
+        self.cached_catalog_metadata = {
```

Comment:
Shouldn't this be the `self.catalog_metadata` you allocated in ``__init__``? Also, since this function currently returns a static dictionary, do we even need `self.catalog_metadata` to hold onto it?

Reply:
Now using a local dict.

Reply:
deleted self.catalog_metadata

Reply:
I still see usage of `self.catalog_metadata` in this method

Reply:
A local dictionary is now assigned and returned